relayer/validate.go

```diff
@@ -0,0 +1,127 @@
+package relayer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	"github.com/athanorlabs/atomic-swap/coins"
+	"github.com/athanorlabs/atomic-swap/common/types"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	"github.com/athanorlabs/atomic-swap/net/message"
+)
+
+func validateClaimRequest(
+	ctx context.Context,
+	request *message.RelayClaimRequest,
+	ec *ethclient.Client,
+	ourSFContractAddr ethcommon.Address,
+) error {
+	err := validateClaimValues(ctx, request, ec, ourSFContractAddr)
+	if err != nil {
+		return err
+	}
+
+	return validateClaimSignature(ctx, ec, request)
+}
+
+// validateClaimValues validates the non-signature aspects of the claim request:
+//  1. the claim request swap factory contract is byte compatible with ours
```

Comment:
```suggestion
//  1. the claim request swap factory contract bytecode matches ours
```

Reply:
Done. I also combined (1) and (2) into the same bullet. (See below.)